### ISSUE TITLE
In CheckedListBox of SelectCheckboxMenu the square of empty item is overloaded by next item

### ISSUE THREAD

**Issue by schlebe** [primefaces/primefaces#967]
The EMPTY item in the listbox of SelectCheckBoxMenu is not displayed correctly

![image of squareoverloaded](http://i.imgur.com/TgpV3c5s.jpg)

Like you can see, square of empty item is overloaded by next item !

I have written a correction for this little UI mistake.

In file org\primefaces\component\selectcheckboxmenu\SelectCheckboxMenuRenderer.java

In method encodeOption()

I have replace 

``` java
        if(escaped)
            writer.writeText(option.getLabel());
        else
            writer.write(option.getLabel());
```

by

``` java
        //2015-12-10:BS: height of item SPACES in Menu ListBox is not correct !
        //<UPDATE> 
        String sLabel = option.getLabel();
        if(escaped)
            {
            if (sLabel.isEmpty())
                {
                if (sLabel.isEmpty()) sLabel = "&nbsp;";
                writer.write(sLabel);
                }
            else
                {
                writer.writeText(sLabel,null);
                }
            }
        else
            {
            if (sLabel.isEmpty()) sLabel = "&nbsp;";
            writer.write(sLabel);
            }
        //</UPDATE> 

```

With this correction, PrimeFaces display ...

![image of rightsquare](http://i.imgur.com/WU1u0O8s.jpg)

I have used the same tip used by PrimeFaces team (using &nbsp;) for the first CheckBox that allows to select all items.

Can somebody by Primefaces implement this solution or can I do it myself ?

It's the first time that I do that and I need a little help to follow your procedure.

**Comment by sudheerj** [comment]
You can create a pull request instead with little changes

**Comment by schlebe** [comment]
I have do that. Now I wait a response.

**Comment by schlebe** [comment]
but this issue is still OPEN

**Comment by mertsincan** [comment]
Thanks a lot for your pull request. Fixed for 5.3.5, 5.2.18 and 6.0.
